### PR TITLE
Let's work together

### PR DESCRIPTION
**Pull request by moroine**
Hi,

I'm the maintainer of the main repository you've forked. I'm starting to work again on that repository ! So I would like to know if you're interested by merging back your features in the main repository ?

I figured out we develop quite the same thing since v0.3.0. I tried to integrate most of your changes into the main repository. If you could have a look at it and discuss about merging back the fork into one repository it would be awesome. 

Thank you,

### CODE REVIEW

**Comment by moroine** [comment]
Btw the diff in github seems to be invalid, this is not the reality.

**Comment by GDXbsv** [comment]
Hello. It is a great idea. I will look inside your commits.

File: src/DependencyInjection/AdactiveSasSaml2BridgeExtension.php [hunk 1]
```diff
@@ -30,6 +30,8 @@ public function load(array $configs, ContainerBuilder $container)
         $loader = new Loader\XmlFileLoader($container, new FileLocator(__DIR__.'/../Resources/config'));
         $loader->load('services.xml');
 
+        var_dump($config);
```
Comment (GDXbsv) [inline src/DependencyInjection/AdactiveSasSaml2BridgeExtension.php]: Please remove var_dump

File: src/Entity/HostedEntities.php [hunk 2]
```diff
@@ -88,6 +88,9 @@ public function getIdentityProvider()
         $configuration['logoutUrl'] = $this->generateUrl(
             $this->identityProviderConfiguration['logout_route']
         );
+
+        var_dump($this->identityProviderConfiguration);
```
Comment (GDXbsv) [inline src/Entity/HostedEntities.php]: Please remove var_dump

File: src/DependencyInjection/AdactiveSasSaml2BridgeExtension.php (hunk 1 above)
Comment (GDXbsv) [inline src/DependencyInjection/AdactiveSasSaml2BridgeExtension.php]: Please remove var_dump

File: src/SAML2/Provider/HostedIdentityProviderProcessor.php [hunk 3]
```diff
@@ -253,8 +280,15 @@ public function processSingleSignOn(Request $httpRequest)
         $inputBinding = $this->bindingContainer->get($this->identityProvider->getSsoBinding());
 
         try {
-            $authRequest = $inputBinding->receiveSignedAuthnRequest($httpRequest);
+            if($this->identityProvider->wantSignedAuthnRequest()){
```
Comment (GDXbsv) [inline src/SAML2/Provider/HostedIdentityProviderProcessor.php:283]: For example: Slack send me SignedAuthn Request, FreshDesk send me Unsigned AuthnRequest. I want to decide it according to SP config.
My way is:
```
$authRequest = $inputBinding->receiveAuthnRequest($httpRequest);
            $sp = $this->getServiceProvider($authRequest->getIssuer());
            if ($sp->wantSignedAuthnRequest()) {
                $authRequest = $inputBinding->receiveSignedAuthnRequest($httpRequest);
            }

            $this->validateRequest($authRequest);
```

File: src/DependencyInjection/Configuration.php [hunk 4]
```diff
@@ -65,6 +83,16 @@ private function addHostedSection(ArrayNodeDefinition $node)
                                 ->scalarNode('private_key')
                                     ->info('The absolute path to the private key used to sign Responses to AuthRequests with')
                                 ->end()
+                                ->arrayNode('signing')
```
Comment (GDXbsv) [inline src/DependencyInjection/Configuration.php:86]: I would like to have these options only in the one place. In the SP config. When these in two places I am little confusing.

File: src/SAML2/Provider/HostedIdentityProviderProcessor.php [hunk 5]
```diff
@@ -329,13 +373,18 @@ public function continueSingleSignOn()
...
      * @return \Symfony\Component\HttpFoundation\Response
+     * @throws \AdactiveSas\Saml2BridgeBundle\Exception\InvalidArgumentException
      */
     public function processSingleLogoutService(Request $httpRequest)
     {
         $inputBinding = $this->bindingContainer->get($this->identityProvider->getSlsBinding());
 
         try {
-            $logoutMessage = $inputBinding->receiveSignedMessage($httpRequest);
+            if($this->identityProvider->wantSignedLogoutRequest()){
```
Comment (GDXbsv) [inline src/SAML2/Provider/HostedIdentityProviderProcessor.php:383]: Why we get this parameter not from the SP?

Comment (GDXbsv) [reply src/SAML2/Provider/HostedIdentityProviderProcessor.php:383]: I would prefer 
```
            $logoutMessage = $inputBinding->receiveUnsignedMessage($httpRequest);
            $sp = $this->getServiceProvider($logoutMessage->getIssuer());
            if ($sp->wantSignedLogoutRequest()) {
                $logoutMessage = $inputBinding->receiveSignedMessage($httpRequest);
            }
```

**Review by GDXbsv — COMMENTED** [review]
I have merged those changes into the my repository. And I have created PR https://github.com/AdactiveSAS/saml2-bridge-bundle/pull/8 to merge all my changes into the your repository.

File: src/SAML2/Provider/HostedIdentityProviderProcessor.php (hunk 5 above)
Comment (moroine) [reply src/SAML2/Provider/HostedIdentityProviderProcessor.php:383]: First of all, I told myself that is the idp to require or not signing requests. But this is too strict, by let this being configured on SP, this is much more flexible

File: src/DependencyInjection/Configuration.php (hunk 4 above)
Comment (moroine) [reply src/DependencyInjection/Configuration.php:86]: Yeah ! I totally agree !

**Pull request merged**